internal/tools/aws/database_migration.go

```diff
@@ -0,0 +1,132 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/rds"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/mattermost/mattermost-cloud/model"
+)
+
+// RDSDatabaseMigration is a migrated database backed by AWS RDS.
+type RDSDatabaseMigration struct {
+	awsClient            *Client
+	masterInstallationID string
+	slaveInstallationID  string
+}
+
+// NewRDSDatabaseMigration returns a new RDSDatabaseMigration.
+func NewRDSDatabaseMigration(masterInstallationID, slaveInstallationID string, awsClient *Client) *RDSDatabaseMigration {
+	return &RDSDatabaseMigration{
+		awsClient:            awsClient,
+		masterInstallationID: masterInstallationID,
+		slaveInstallationID:  slaveInstallationID,
+	}
+}
+
+// Setup sets access from one RDS database to another and sets any configuration needed for replication.
+func (d *RDSDatabaseMigration) Setup(logger log.FieldLogger) (string, error) {
+	masterInstanceSG, err := d.describeDBInstanceSecurityGroup(RDSMasterInstanceID(d.masterInstallationID))
+	if err != nil {
+		return "", err
+	}
+
+	slaveInstanceSG, err := d.describeDBInstanceSecurityGroup(RDSMigrationInstanceID(d.slaveInstallationID))
+	if err != nil {
+		return "", err
+	}
+
+	_, err = d.awsClient.Service().ec2.AuthorizeSecurityGroupIngress(&ec2.AuthorizeSecurityGroupIngressInput{
+		GroupId: masterInstanceSG.GroupId,
+		IpPermissions: []*ec2.IpPermission{
+			{
+				FromPort:   aws.Int64(3306),
+				IpProtocol: aws.String("tcp"),
+				ToPort:     aws.Int64(3306),
+				UserIdGroupPairs: []*ec2.UserIdGroupPair{
+					{
+						Description: aws.String("Ingress Traffic from other RDS instance"),
+						GroupId:     slaveInstanceSG.GroupId,
+					},
+				},
+			},
+		},
+	})
+	if err != nil && !strings.Contains(err.Error(), "already exists") {
+		return "", err
+	}
+
+	logger.WithField("migration-installation-id", d.masterInstallationID).Info("database migration setup completed")
```

Comment:
Nit: Capital letter start for log messages
```suggestion
	logger.WithField("migration-installation-id", d.masterInstallationID).Info("Database migration setup completed")
```

Reply:
Good call! I think I saw how to do that somewhere.